Content.Shared/Nutrition/EntitySystems/IngestionSystem.API.cs

```diff
@@ -0,0 +1,364 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Components.SolutionManager;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.EntityEffects.Effects;
+using Content.Shared.FixedPoint;
+using Content.Shared.Inventory;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Nutrition.Prototypes;
+using Content.Shared.Verbs;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Nutrition.EntitySystems;
+
+/// <summary>
+/// Public API for Ingestion System so you can build your own form of ingestion system.
+/// </summary>
+public sealed partial class IngestionSystem
+{
+    // List of prototypes that other components or systems might want.
+    public static readonly ProtoId<EdiblePrototype> Food = "Food";
+    public static readonly ProtoId<EdiblePrototype> Drink = "Drink";
+
+    public const float MaxFeedDistance = 1.0f; // We should really have generic interaction ranges like short, medium, long and use those instead...
+    // BodySystem has no way of telling us where the mouth is so we're making some assumptions.
+    public const SlotFlags DefaultFlags = SlotFlags.HEAD | SlotFlags.MASK;
+
+    #region Ingestion
+
+    /// <summary>
+    /// An entity is trying to ingest another entity in Space Station 14!!!
+    /// </summary>
+    /// <param name="user">The entity who is eating.</param>
+    /// <param name="ingested">The entity that is trying to be ingested.</param>
+    /// <param name="ingest">Whether we're actually ingesting the item or if we're just testing. If false, treat as "CanIngest"</param>
+    /// <returns>Returns true if we can ingest the item.</returns>
+    public bool TryIngest(EntityUid user, EntityUid ingested, bool ingest = true)
```

Comment:
get bonked for an insane API. Just make a separate CanIngest method rather than passing in a bool for what _should_ be the regular behavior of this function